sdk/communication/Azure.Communication.Administration/tests/PhoneNumberAdministrationClient/PhoneNumberAdministrationClientLiveTests.cs

```diff
@@ -0,0 +1,233 @@
+
+        [Test]
+        [TestCase(null)]
+        [TestCase("en-US")]
+        public async Task GetAreaCodesForPlan(string? locale)
+        {
+            var client = CreateClient();
+            var countryCode = "US";
+
+            var pageablePhonePlanGroups = client.GetPhonePlanGroupsAsync(countryCode, locale);
+            var phonePlanGroups = await pageablePhonePlanGroups.ToEnumerableAsync().ConfigureAwait(false);
+
+            string phonePlanGroupId = phonePlanGroups.First(group => group.PhoneNumberType == PhoneNumberType.Geographic).PhonePlanGroupId;
+            var pageablePhonePlans = client.GetPhonePlansAsync(countryCode, phonePlanGroupId, locale);
+            var phonePlans = await pageablePhonePlans.ToEnumerableAsync();
+            var phonePlanId = phonePlans.First().PhonePlanId;
+
+            var locationOptionsResponse = await client.GetPhonePlanLocationOptionsAsync(countryCode, phonePlanGroupId, phonePlanId).ConfigureAwait(false);
+
+            var locationOptions = new List<LocationOptionsQuery>
+            {
+                new LocationOptionsQuery
+                {
+                    LabelId = "state",
+                    OptionsValue = "WA"
+                },
+                new LocationOptionsQuery
+                {
+                    LabelId = "city",
+                    OptionsValue = "NOAM-US-WA-SE"
+                }
+            };
+
+            var areaCodes = await client.GetAllAreaCodesAsync("selection", countryCode, phonePlanId, locationOptions);
+
+            Assert.IsNotNull(areaCodes.Value.PrimaryAreaCodes);
+            Assert.IsTrue(areaCodes.Value.PrimaryAreaCodes.Count > 0);
+        }
+
+        [Test]
+        [TestCase(null)]
+        [TestCase("en-US")]
+        [AsyncOnly]
+        public async Task CreateReservationErrorState(string? locale)
+        {
+            var client = CreateClient();
+            var countryCode = "US";
+
+            var pageablePhonePlanGroups = client.GetPhonePlanGroupsAsync(countryCode, locale);
+            var phonePlanGroups = await pageablePhonePlanGroups.ToEnumerableAsync().ConfigureAwait(false);
+
+            string phonePlanGroupId = phonePlanGroups.First(group => group.PhoneNumberType == PhoneNumberType.TollFree).PhonePlanGroupId;
+            var pageablePhonePlans = client.GetPhonePlansAsync(countryCode, phonePlanGroupId, locale);
+            var phonePlan = (await pageablePhonePlans.ToEnumerableAsync()).First();
+            var areaCode = phonePlan.AreaCodes.First();
+
+            var reservationOptions = new CreateReservationOptions("My reservation", "my description", new[] { phonePlan.PhonePlanId }, areaCode);
+            reservationOptions.Quantity = 1;
+            var reservationOperation = await client.StartReservationAsync(reservationOptions);
+
+            try
+            {
+                await reservationOperation.WaitForCompletionAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
```

Comment:
I'm not following. Why do we expect this to fail and how would this test continue past line 182?

Reply:
I have made a recording of operation that became an error state. This is the case to test we handling error state properly. There are sore asserts after the catch that are redundant. I will remove them